src/java.base/share/classes/java/util/HexFormat.java

```diff
@@ -0,0 +1,1021 @@
+    }
+
+    /**
+     * Returns a copy of this {@code HexFormat} with the {@code delimiter}.
+     * @param delimiter the {@code delimiter}, non-null, may be empty
+     * @return a copy of this {@code HexFormat} with the {@code delimiter}
+     */
+    public HexFormat withDelimiter(String delimiter) {
+        return new HexFormat(delimiter, this.prefix, this.suffix, this.digits);
+    }
+
+    /**
+     * Returns a copy of this {@code HexFormat} with the {@code prefix}.
+     *
+     * @param prefix a prefix, non-null, may be empty
+     * @return a copy of this {@code HexFormat} with the {@code prefix}
+     */
+    public HexFormat withPrefix(String prefix) {
+        return new HexFormat(this.delimiter, prefix, this.suffix, this.digits);
+    }
+
+    /**
+     * Returns a copy of this {@code HexFormat} with the {@code suffix}.
+     *
+     * @param suffix a {@code suffix}, non-null, may be empty
+     * @return a copy of this {@code HexFormat} with the {@code suffix}
+     */
+    public HexFormat withSuffix(String suffix) {
+        return new HexFormat(this.delimiter, this.prefix, suffix, this.digits);
+    }
+
+    /**
+     * Returns a copy of this {@code HexFormat} to use uppercase hex characters.
+     * The uppercase hexadecimal characters are {@code "0-9", "A-F"}.
+     *
+     * @return a copy of this {@code HexFormat} with uppercase hexadecimal characters
+     */
+    public HexFormat withUppercase() {
+        return new HexFormat(this.delimiter, this.prefix, this.suffix, HexFormat.UPPERCASE_DIGITS);
+    }
+
+    /**
+     * Returns a copy of this {@code HexFormat} to use lowercase hex characters.
+     * The lowercase hexadecimal characters are {@code "0-9", "a-f"}.
+     *
+     * @return a copy of this {@code HexFormat} with lowercase hex characters
+     */
+    public HexFormat withLowercase() {
+        return new HexFormat(this.delimiter, this.prefix, this.suffix, HexFormat.LOWERCASE_DIGITS);
+    }
+
+    /**
+     * Returns the {@code delimiter} between hexadecimal values in a formatted byte array.
+     *
+     * @return return the {@code delimiter}, non-null, may be empty {@code ""}
+     */
+    public String delimiter() {
+        return delimiter;
+    }
+
+    /**
+     * Returns the {@code prefix} used for each hexadecimal value in a formatted byte array.
+     *
+     * @return returns the {@code prefix}
+     */
+    public String prefix() {
+        return prefix;
+    }
+
+    /**
+     * Returns the {@code suffix} used for each hexadecimal value in a formatted byte array.
+     *
+     * @return returns the {@code suffix}
+     */
+    public String suffix() {
+        return suffix;
+    }
+
+    /**
+     * Returns {@code true} if the hexadecimal digits will be uppercase,
+     *          otherwise {@code false}.
+     * @return returns {@code true} if the hexadecimal digits will be uppercase,
+     *          otherwise {@code false}
+     */
```

Comment:
I don't think it would be called HexFormat doing Hexadecimal conversions if it could be an arbitrary set of digits.
The Arrays.equals implementation contains the == test so its implementers choice.